File: Common/InMotionGIT.Common - Copy/Helpers/SessionHandler.vb
Imports System.Web

Namespace Helpers

    Public Class SessionHandler

        ''' <summary>
        ''' Almacena objetos en la sección.
        ''' </summary>
        ''' <param name="key">Nombre con que se almacena el objeto</param>
        ''' <param name="value">Objeto a almacenar</param>
        Public Shared Sub Save(key As String, value As Object)
            Dim session As System.Web.SessionState.HttpSessionState = Nothing
            If HttpContext.Current IsNot Nothing Then
                If HttpContext.Current.Request IsNot Nothing Then
                    session = HttpContext.Current.Session
                End If
            End If

            If session.IsNotEmpty Then
                Dim valueFound As String = (From itemKey In session.Keys Where itemKey.ToString.Equals(key) Select itemKey).FirstOrDefault
                If valueFound.IsEmpty Then
                    session.Add(key, value)
                Else
                    session(key) = value
                End If
            End If
        End Sub

        ''' <summary>
        '''  Elimina el objetos en la sección.
        ''' </summary>
        ''' <param name="key">Nombre del objeto a eliminar</param>
        Public Shared Sub Remove(key As String)
            Dim session As System.Web.SessionState.HttpSessionState = Nothing
            If HttpContext.Current IsNot Nothing Then
                If HttpContext.Current.Request IsNot Nothing Then
                    session = HttpContext.Current.Session
                End If
            End If

            If session.IsNotEmpty Then
                session.Remove(key)
            End If
        End Sub

        ''' <summary>
        ''' Retorna el objeto en la sección.
        ''' </summary>
        ''' <param name="key">Nombre con que se almacena el objeto</param>
        Public Shared Function Retrieve(key As String) As Object
            Dim result As Object = Nothing
            Dim session As System.Web.SessionState.HttpSessionState = Nothing
            If HttpContext.Current IsNot Nothing Then
                If HttpContext.Current.Request IsNot Nothing Then
                    session = HttpContext.Current.Session
                End If
            End If

            If session.IsNotEmpty Then
                Dim valueFound As String = (From itemKey In session.Keys Where itemKey.ToString.Equals(key) Select itemKey).FirstOrDefault
                If valueFound.IsNotEmpty Then
                    result = session(key)
                End If
            End If
            Return result
        End Function

        ''' <summary>
        ''' Verifica si existe el key de un objeto en sección
        ''' </summary>
        ''' <param name="key">Nombre a verificar la existencia en el sección</param>
        ''' <returns></returns>
        Public Shared Function Exist(key As String) As Boolean
            Dim result As Boolean = False
            Dim session As System.Web.SessionState.HttpSessionState = Nothing
            If HttpContext.Current IsNot Nothing Then
                If HttpContext.Current.Request IsNot Nothing Then
                    session = HttpContext.Current.Session
                    If session.IsNotEmpty Then
                        Dim valueFound As String = (From itemKey In session.Keys
                                                    Where itemKey.ToString.Equals(key)
                                                    Select itemKey).FirstOrDefault
                        If valueFound.IsNotEmpty Then
                            Return True
                        Else
                            Return False
                        End If
                    Else
                        Return False
                    End If
                Else
                    Return False
                End If
            Else
                Return False
            End If
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/SessionHandler.cs
using InMotionGIT.Common.Extensions;
using System.Linq;
using System.Web;

namespace InMotionGIT.Common.Helpers
{
    public class SessionHandler
    {
        /// <summary>
        /// Almacena objetos en la sección.
        /// </summary>
        /// <param name="key">Nombre con que se almacena el objeto</param>
        /// <param name="value">Objeto a almacenar</param>
        public static void Save(string key, object value)
        {
            System.Web.SessionState.HttpSessionState session = null;
            if (HttpContext.Current is not null)
            {
                if (HttpContext.Current.Request is not null)
                {
                    session = HttpContext.Current.Session;
                }
            }

            if (session.IsNotEmpty())
            {
                string valueFound = HttpContext.Current.Session.Keys.Cast<string>().FirstOrDefault(itemKey => itemKey.Equals(key));
                if (valueFound.IsEmpty())
                {
                    session.Add(key, value);
                }
                else
                {
                    session[key] = value;
                }
            }
        }

        /// <summary>
        /// Elimina el objetos en la sección.
        /// </summary>
        /// <param name="key">Nombre del objeto a eliminar</param>
        public static void Remove(string key)
        {
            System.Web.SessionState.HttpSessionState session = null;
            if (HttpContext.Current is not null)
            {
                if (HttpContext.Current.Request is not null)
                {
                    session = HttpContext.Current.Session;
                }
            }

            if (session.IsNotEmpty())
            {
                session.Remove(key);
            }
        }

        /// <summary>
        /// Retorna el objeto en la sección.
        /// </summary>
        /// <param name="key">Nombre con que se almacena el objeto</param>
        public static object Retrieve(string key)
        {
            object result = null;
            System.Web.SessionState.HttpSessionState session = null;
            if (HttpContext.Current is not null)
            {
                if (HttpContext.Current.Request is not null)
                {
                    session = HttpContext.Current.Session;
                }
            }

            if (session.IsNotEmpty())
            {
                string valueFound = HttpContext.Current.Session.Keys.Cast<string>().FirstOrDefault(itemKey => itemKey.Equals(key));
                if (valueFound.IsNotEmpty())
                {
                    result = session[key];
                }
            }
            return result;
        }

        /// <summary>
        /// Verifica si existe el key de un objeto en sección
        /// </summary>
        /// <param name="key">Nombre a verificar la existencia en el sección</param>
        /// <returns></returns>
        public static bool Exist(string key)
        {
            bool result = false;
            System.Web.SessionState.HttpSessionState session = null;
            if (HttpContext.Current is not null)
            {
                if (HttpContext.Current.Request is not null)
                {
                    session = HttpContext.Current.Session;
                    if (session.IsNotEmpty())
                    {
                        string valueFound = HttpContext.Current.Session.Keys.Cast<string>().FirstOrDefault(itemKey => itemKey.Equals(key));
                        if (valueFound.IsNotEmpty())
                        {
                            return true;
                        }
                        else
                        {
                            return false;
                        }
                    }
                    else
                    {
                        return false;
                    }
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }
    }
}